core/src/main/java/org/elasticsearch/action/delete/DeleteResponse.java

```diff
@@ -36,43 +36,38 @@
  */
 public class DeleteResponse extends DocWriteResponse {
 
-    private boolean found;
-
     public DeleteResponse() {
 
     }
 
     public DeleteResponse(ShardId shardId, String type, String id, long version, boolean found) {
-        super(shardId, type, id, version);
-        this.found = found;
+        super(shardId, type, id, version, toOperation(found));
     }
 
+    public static Operation toOperation(boolean found) {
+        return found ? Operation.DELETE: Operation.NOOP;
+    }
 
     /**
      * Returns <tt>true</tt> if a doc was found to delete.
      */
     public boolean isFound() {
-        return found;
+        return operation == Operation.DELETE;
     }
 
     @Override
     public void readFrom(StreamInput in) throws IOException {
```

Comment:
Since it is defined on the superclass it'll still be there, just not a delegates-to-superclass override.